### ISSUE TITLE
Limitations on the number of traits in metagenome data using melonnpan.train

### ISSUE THREAD

**Issue by erawijantari** [biobakery/melonnpan#10]
Hi @himelmallick, thank's for updating the tutorial, it's so helpful.
I tried to re-install melonnpan and rerun the analysis using the same format as given in the tutorial. By doing this, I did not encounter the first error I reported in #4 , thank's.
However, I still get the same as the second error that I reported in #4.
Previously you suggest me to carefully remove the metagenomic features with only a handful of unique values, which I am not sure if I can understand correctly.
So, I filter the metagenome traits based on its prevalence and relative abundance that hopefully remove those with unique values. 
Before the filtering the number of traits were 1,282,446 genes and I cut it down to 799,450 genes after filtering.
The metabolome traits were 340 metabolites.
I run melonnpan using both (with and without genes filtering) and both produce this error
```
Error in ztransform(formula, data, family) : trait is binary
Calls: <Anonymous> -> apply -> FUN -> ztransform
Execution halted
```

Then I tried to randomly select 100 traits from metagenome and it runs perfectly fine.
My question is, is there any limitations in the number of genes that should be included using the melonnpan.train?
For your informations, the metagenome data did not produce the same error when I use the melonnpan.predict. 
Thank's in advance.

**Comment by himelmallick** [comment]
Hi there, @erawijantari - as far as I can tell, MelonnPan should work with any number of features but it is recommended that you retain only those features that are either sufficiently prevalent and abundant or variable across samples. This greatly reduces the computational burden in the fitting stage. But as far as your error is concerned, it looks like, some features in your metagenomic data have only a few unique values. You can easily do a sanity check using something like https://stackoverflow.com/questions/22196078/count-unique-values-for-every-column. Could you please verify the minimum number of unique values per feature is at least greater than 5? Many thanks!

**Comment by EllieYin567** [comment]
@erawijantari @himelmallick I also had this problem before and my experience is that for some of the gene/metabolite features, you only have 2 unique values, that is why for "ztransform", a "trait binary" error would occur. After removing these features, the error went away in my case. Hope it helps!

**Comment by erawijantari** [comment]
Hi @himelmallick sorry for late responds and thank's for suggestions.
I did the sanity check that you've suggested and my tables indeed have features with unique values less than 5, so I tried to remove those features and the error went away. 
And thank's @EllieYin567 for your helpful comment, I also tried to remove the features with only have 2 unique values and yes, the error went away as well.
I will try to compare the results I got.
Thank you!
